[APPSEC-11122] Add documentation (English) for trace qualification (#20113)
DataDog/documentation: content/en/security/application_security/threats/trace_qualification.md

This is where trace qualification comes in - ASM will evaluate the conditions in which each trace was generated, and - when possible - label it as harmful or safe.

For example, a service that doesn't use SQL databases can not be harmed with a SQL injection. For this service, ASM will qualify SQL injection attacks as unsuccessful (*None successful* in the user interface).

## Possible qualification outcomes

{{< img src="security/application_security/threats/trace_qualification/trace-qualification-traces.png" alt="ASM trace list with the qualification facet showing the possible qualification results">}}

There are four types of qualification outcomes and you can use the facet menu to find the corresponding traces:

| Qualification result | Description |
|------|-------------|
| `Unknown` | ASM has qualification rules for this attack, but did not have enough information to make a qualification decision. |
| *No value* | ASM does not have qualification rules for this type of attack. |

> | `None successful` | ASM determined that no attacks in this trace could be harmful. |

Reviewer: ```suggestion
| `None successful` | ASM determined that attacks in this trace were not harmful. |
```
Tweaking the phrasing so that `not` is closer to `harmful`. Makes it a bit easier to parse
